src/src/com/microsoft/aad/adal/DefaultTokenCacheStore.java

```diff
@@ -367,4 +370,78 @@ public boolean contains(String key) {
+                tokenItem = tokenCacheItem;
+                //return the cacheItem which contains the FRT for this userId
+                //there should be at most only one FRT record for each userId
+                return tokenItem;
+            }
+        }
+        //If the cache does not contain the FRT for this user return null
+        Logger.i(TAG, "Cannot find the family refresh token cache item for this user in cache", "");
+        return null;
+    }
+    
+    protected String serialize(String uniqueUserId) throws UsageAuthenticationException {
+        if (StringExtensions.IsNullOrBlank(uniqueUserId)) {
+            throw new IllegalArgumentException("uniqueUserId");
+        }
+        
+        if((new BrokerProxy(mContext)).canSwitchToBroker()){
+            throw new UsageAuthenticationException(ADALError.FAIL_TO_EXPORT,"Failed to export the FID because broker is enabled.");
+        } else {
+            TokenCacheItem tokenItem = getFamilyRefreshTokenItemForUser(uniqueUserId);
+            
+            if(tokenItem == null) {
+                Logger.i(TAG, "Cannot serialize the FRT cache item: family refresh token is null", "");
+                return null;
+            }
+
+            Gson gson = new GsonBuilder()
+                    .registerTypeAdapter(BlobContainer.class, new BlobContainerAdapter())
+                    .create();
+            BlobContainer blobContainer = new BlobContainer(tokenItem);
+            Logger.i(TAG, "family refresh token cache item:" + blobContainer.toString(), "");
+            String json = gson.toJson(blobContainer);
+            return json;                
+        }      
+    }
+    
+    protected void deserialize(String serializedBlob) throws AuthenticationException{
+        if (StringExtensions.IsNullOrBlank(serializedBlob)) {
+            throw new IllegalArgumentException("serializedBlob");
+        }
+            
+        if((new BrokerProxy(mContext)).canSwitchToBroker()) {
+            throw new UsageAuthenticationException(ADALError.FAIL_TO_IMPORT,"Failed to import the serialized blob because broker is enabled.");
+        } else {
+            Gson gson = new GsonBuilder()
+                    .registerTypeAdapter(BlobContainer.class, new BlobContainerAdapter())
+                    .create(); 
+            BlobContainer blobContainer;
+            
+            try {
+                blobContainer = gson.fromJson(serializedBlob, BlobContainer.class);
+            } catch (JsonParseException exp) {
+                throw new DeserializationAuthenticationException(ADALError.FAIL_TO_DESERIALIZE,
+                        exp.getMessage());
+            }
+            
+            Logger.i(TAG + ".deserialize", "Json parsing for serializedBlob :" + blobContainer.toString(), "");
+            TokenCacheItem tokenCacheItem = blobContainer.getTokenItem();
+            String cacheKey = CacheKey.createCacheKey(tokenCacheItem);
```

Comment:
Did the key structure change? I think it was (A, U, C, R) before. Could you send me the wiki or PR or any doc I could have a look?